theme/theme.inc

```diff
@@ -303,7 +306,18 @@ function islandora_solr_metadata_query_fields($object, &$solr_fields) {
           // Replace value with updated field display.
           $field['value'] = $updated_value;
         }
-        unset($field);
+
+        // Apply rules configurations for this field as configured
+        // on the metadata configuration page. Unset the field from
+        // the $solr_fields array to prevent it from being shown.
+        $permission = drupal_array_get_nested_value($field, array('permissions'));
+        if ($permission['enable_permissions'] == 1) {
+          foreach ($user->roles as $rid => $role_name) {
+            if (!in_array($rid, $permission['permissions'])) {
+              unset($solr_fields[$field['solr_field']]);
```

Comment:
I'm not sure this is applying permissions the desired way... This would require _all_ roles which a user has to have permission to view the field, when we probably want to allow if _any_ of the user's roles has permission... So, something like:
```php
if ($permission['enable_permissions'] && !array_intersect(array_keys($user->roles), $permission['permissions'])) {
  // Permissions are being enforced and none of our roles are allowed...
  unset($solr_fields[$field['solr_field']]);
}
```

Or if we really want to stick with iteration, would be something like:
```php
if ($permission['enable_permissions']) {
  $remove = TRUE;
  foreach ($user->roles as $rid => $role_name) {
    if (in_array($rid, $permission['permissions'])) {
      $remove = FALSE;
      break;
    }
  }
  if ($remove) {
    unset($solr_fields[$field['solr_field']]);
  }
}
```